pyonfx/utils.py

```diff
@@ -163,32 +165,32 @@ class FrameUtility:
 
     """
 
-    def __init__(self, start_time: float, end_time: float, fr: float = 41.71):
+    def __init__(self, start_time: int, end_time: int, fps: Fraction = Fraction(24000, 1001)):
         # Checking for invalid values
-        if start_time < 0 or end_time < 0 or fr <= 0 or end_time < start_time:
+        if start_time < 0 or end_time < 0 or fps <= 0 or end_time < start_time:
             raise ValueError("Positive values and/or end_time > start_time expected.")
 
-        self.start_time = self.current_time = start_time
-        self.end_time = end_time
-        self.fr = self.fr_time = fr
+        self.fps = fps
+
+        self.start_time_image = self.current_time = Convert.ms2frame(start_time, self.fps, "start")
+        self.end_time_image = Convert.ms2frame(end_time, self.fps, "end")
+
+        self.n = self.end_time_image - self.start_time_image + 1
+
 
-        # Calculating number of frames
-        self.n = math.ceil((end_time - start_time) / fr)
 
     def __iter__(self):
-        # Compute values for the first n-1 frames
-        for i in range(1, self.n):
+        for i in range(self.n):
+
             yield (
-                round(self.current_time, 2),
-                round(self.current_time + self.fr, 2),
-                i,
+                Convert.frame2ms(self.current_time, self.fps, "start"),
+                Convert.frame2ms(self.current_time, self.fps, "end"),
+                i + 1,
                 self.n,
             )
-            self.current_time += self.fr
-            self.fr_time += self.fr
 
-        # Compute values for the last frame, clamping the end_time of the frame at self.end_time
-        yield (round(self.current_time, 2), round(self.end_time, 2), self.n, self.n)
+            self.current_time = self.current_time + 1
```

Comment:
Instead of changing this part, can't we just use the bound function at the beginning and then keep working with ms as it was before?

Reply:
Which value will you use to make to get the next frame?

In frame, it is easy. If you have 1000. The next frame is 1001.

In ms, it is not that simple.

Reply:
It would need to be properly tested.

But, has frameutility "play" with frame, I think all calculations should be made in image